### PR TITLE
Add stash support for mysqldatabase

### PR DESCRIPTION
**Pull request by TasdidurRahman**
Signed-off-by: TasdidurRahman <tasdid@appscode.com>

### CODE REVIEW

File: pkg/controllers/schema/mysqldatabase_controller.go [hunk 1]
```diff
@@ -46,8 +47,8 @@ import (
 // MySQLDatabaseReconciler reconciles a MySQLDatabase object
 type MySQLDatabaseReconciler struct {
 	client.Client
-	Scheme           *runtime.Scheme
-	AppcatalogClient appcatCs.Interface
+	Scheme *runtime.Scheme
+	//AppcatalogClient appcatCs.Interface
```
Comment (spectro30) [inline pkg/controllers/schema/mysqldatabase_controller.go:51]: remove commented code if it's not required(for now or future)

File: pkg/controllers/schema/mysqldatabase_controller.go [hunk 2]
```diff
@@ -158,23 +164,35 @@ func (r *MySQLDatabaseReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, err
 	}
 
+	if !kmapi.IsConditionTrue(schemaObj.Status.Conditions, string(schemaapi.DatabaseSchemaConditionTypeSecretAccessRequestReady)) {
```
Comment (spectro30) [inline pkg/controllers/schema/mysqldatabase_controller.go:167]: why don't merge this if block with next one, as they are returning same things.

File: pkg/controllers/schema/mysqldatabase_controller_helper.go [hunk 3]
```diff
@@ -511,21 +756,35 @@ func (r *MySQLDatabaseReconciler) MySQLRoleTask(ctx context.Context, schemaObj *
...
+	if kmapi.IsConditionTrue(schemaObj.Status.Conditions, string(schemaapi.DatabaseSchemaConditionTypeRoleReady)) {
 		return true, nil
 	}
 	return false, nil
 }
 
 func (r *MySQLDatabaseReconciler) AccessRequestTask(ctx context.Context, schemaObj *schemaapi.MySQLDatabase) (bool, error) {
 
-	curCond := &schemaObj.Status.Conditions
+	//curCond := &schemaObj.Status.Conditions
```
Comment (spectro30) [inline pkg/controllers/schema/mysqldatabase_controller_helper.go:787]: remove?

**Review by spectro30 — CHANGES_REQUESTED** [review]

File: pkg/controllers/schema/mysqldatabase_controller.go (hunk 1 above)
Comment (TasdidurRahman) [reply pkg/controllers/schema/mysqldatabase_controller.go]: done

File: pkg/controllers/schema/mysqldatabase_controller.go (hunk 2 above)
Comment (TasdidurRahman) [reply pkg/controllers/schema/mysqldatabase_controller.go]: merged conditions

**Review by spectro30 — COMMENTED** [review]
LGTM

**Pull request merged**